h2o-algos/src/main/java/hex/coxph/CoxPHModel.java

```diff
@@ -433,23 +435,7 @@ public CoxPHMojoWriter getMojo() {
 
   @Override
   public boolean haveMojo() {
-    final boolean enabled = super.haveMojo() && hasOnlyNumericInteractions();
-    if (! enabled) {
-      boolean forceEnabled = H2O.getSysBoolProperty("coxph.mojo.forceEnable", false);
-      if (forceEnabled) {
-        Log.warn("Model " + this._key + " doesn't technically support MOJO, but MOJO support was force-enabled.");
-        return true;
-      }
-    }
-    return enabled;
-  }
-
-  boolean hasOnlyNumericInteractions() {
-    if (_output._interactionSpec == null) {
-      return true;
-    }
-    return Stream.of(_output.data_info._interactions)
-            .allMatch(InteractionPair::isNumeric);
+    return true;
```

Comment:
Tomas:  This is so messed up.  Basically the CoxPHMojoWriter will call the modelDescriptor() method to generate this _extraMojoFeatures which is then used to generate the correct field values for columnNames, nfeatures.  If I changed that, I will have to generate all those stuff myself.  Here is the code: 
![image](https://user-images.githubusercontent.com/7231712/233240853-9c16180f-8b69-4d53-976b-4d9a1d60d43b.png)

Reply:
Here is where I need the column names in CoxPHMojoWriter:

![image](https://user-images.githubusercontent.com/7231712/233241210-5fc9ddd7-68af-4634-8a19-698784835d7e.png)

Reply:
Weird. What I meant was to delete the `hex.coxph.CoxPHModel#haveMojo` so it uses `Model#haveMojo`, which looks like this:
```java
  public boolean haveMojo() {
    if (_parms._preprocessors != null) return false; // until PUBDEV-7799, disable model MOJO if it was trained with embedded TE.
    final String algoName = _parms.algoName();
    return ModelBuilder.getRegisteredBuilder(algoName)
            .map(ModelBuilder::haveMojo)
            .orElseGet(() -> {
              Log.warn("Model Builder for algo = " + algoName + " is not registered. " +
                      "Unable to determine if Model has a MOJO. Please override method haveMojo().");
              return false;
            });
  }
```

So the `hex.coxph.CoxPHModel#haveMojo` would be true unless there is some preprocessor (since it uses value from the model builder (`hex.coxph.CoxPH#haveMojo`) which is `true` (https://github.com/h2oai/h2o-3/blame/d621d2d90c5007ab6593333691ee11e56628d2d2/h2o-algos/src/main/java/hex/coxph/CoxPH.java#L43-L46)).

So the behavior would change only slightly - if there is some preprocessing we would not support mojo (which I don't think we support anyway).

Does it make sense like this or am I missing something?

Reply:
I misundertand what you are suggesting.  Let me fix it.  Thanks!

Reply:
Yes, you did miss something.  You miss the idea that I may jut be crazy.